src/hooks/use-authentication.ts

```diff
@@ -1,17 +1,33 @@
-import { useSession, signIn, signOut, SignOutParams } from 'next-auth/react'
+import {
+	signIn,
+	SignInOptions,
+	signOut,
+	SignOutParams,
+	useSession,
+} from 'next-auth/react'
 import { SessionData, UserData, ValidAuthProviderId } from 'types/auth'
 
-const DEFAULT_PROVIDER_ID = ValidAuthProviderId.CloudIdp
 const AUTH_ENABLED = __config.flags.enable_auth
+const DEFAULT_PROVIDER_ID = ValidAuthProviderId.CloudIdp
+const DEFAULT_SIGN_IN_CALLBACK_URL = '/profile'
 
 /**
  * A minimal wrapper around next-auth/react's `signIn` function. Purpose is to
  * handle invoking the wrapped function with a default value.
  *
  * https://next-auth.js.org/getting-started/client#signin
  */
-const signInWrapper = (provider: ValidAuthProviderId = DEFAULT_PROVIDER_ID) => {
-	return signIn(provider)
+const signInWrapper = (
+	provider: ValidAuthProviderId = DEFAULT_PROVIDER_ID,
+	options: SignInOptions = {}
```

Comment:
```suggestion
	provider?: ValidAuthProviderId = DEFAULT_PROVIDER_ID,
	options?: SignInOptions = {}
```
Should these be optional if you're providing a default?

Reply:
Actually im unsure if that's valid ts what I suggested lol but anyways, still a question 😆

Reply:
Yep! TypeScript only needs one or the other and doesn't like both.

<img width="560" alt="CleanShot 2022-07-29 at 14 37 36@2x" src="https://user-images.githubusercontent.com/43934258/181823670-f2de0f54-b55a-4c96-8b89-9152995c1695.png">